flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -87,10 +87,16 @@ object UserDefinedFunctionUtils {
       // go over all signatures and find one matching actual signature
       .find { curSig =>
       // match parameters of signature to actual parameters
-      actualSignature.length == curSig.length &&
+      (actualSignature.length == curSig.length &&
         curSig.zipWithIndex.forall { case (clazz, i) =>
           parameterTypeEquals(actualSignature(i), clazz)
-        }
+        }) ||
+        // matching the style which last argument is variable, eg. "Type..." "Type*"
+        (actualSignature.length >= curSig.length &&
+          curSig.zipWithIndex.forall { case (clazz, i) =>
+              parameterTypeEquals(actualSignature(i), clazz) ||
+                (i == curSig.length - 1 && clazz.isArray)
```

Comment:
Even if the last parameter type is an array type, it is possible not a varargs method. Such as `public void eval(int[] arrays)` , it is not a varargs method but a general method, we should add a test to check this feature not affect general array method.

The best way to check whether a method is varargs is using `Method.isVarArgs()`.